content/en/blog/2022-08-26-manage-kyverno-policies-as-ocirepositories/index.md

```diff
@@ -0,0 +1,207 @@
+
+This command will install Flux and create necessary files for us and push them into the repository.
+
+Next, we should install Kyverno first. As we do everything GitOps in mind, we’ll also install Kyverno with Flux. In order to do that, we use HelmRepository and HelmRelease resources:
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: HelmRepository
+metadata:
+  name: kyverno
+  namespace: flux-system
+spec:
+  interval: 10m
+  url: https://kyverno.github.io/kyverno/
+---
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  name: kyverno
+  namespace: flux-system
+spec:
+  interval: 10m
+  targetNamespace: kyverno
+  install:
+    createNamespace: true
+  chart:
+    spec:
+      chart: kyverno
+      sourceRef:
+        kind: HelmRepository
+        name: kyverno
+```
+
+Do not forget to check whether everything works fine before moving into the next steps:
+
+```shell
+$ flux get helmreleases kyverno
+NAME    REVISION        SUSPENDED       READY   MESSAGE
+kyverno v2.5.3          False           True    Release reconciliation succeeded
+```
+
+Now, we are ready to create an OCI image to store my Kyverno policies.
+
+> ⚠️  You can find all the code examples in [GitHub](https://github.com/developer-guy/flux-kyverno-policies).
+
+In order to do that, we will clone our repository that holds Kyverno policies and create an OCI image to store them
+
+```shell
+$ git clone https://github.com/developer-guy/my-kyverno-policies.git
+$ cd my-kyverno-policies
+```
+
+> ⚠️ We are expecting that some other team “DevSecOps” is responsible to maintain and publish the policies to GHCR
+
+Next, we will be applying cert-manager policies into the cluster as an example:
+
+```shell
+$ flux push artifact oci://ghcr.io/developer-guy/policies:v1.0.0 \
+  --path="." \
+  --source="$(git config --get remote.origin.url)" \
+  --revision="$(git branch --show-current)/$(git rev-parse HEAD)"
+► pushing artifact to ghcr.io/developer-guy/policies:v1.0.0
+✔ artifact successfully pushed to ghcr.io/developer-guy/policies@sha256:56e853e3c5c02139c840b7f5c89a02f63ede8dc498ed3925a52360032aa49e60
+```
+
+> ⚠️ **Note:** Don't forget to change the values with your own details!
+
+Last but not least, we need to create an OCIRepository source that points out to my OCI image:
+
+```yaml
+---
+apiVersion: source.toolkit.fluxcd.io/v1beta2
+kind: OCIRepository
+metadata:
+  name: kyverno-policies
+  namespace: flux-system
+spec:
+  interval: 5m
+  url: oci://ghcr.io/developer-guy/policies
+  ref:
+    semver: "v1.x"
+  secretRef:
+    name: ghcr-auth
+---
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
+metadata:
+  name: kyverno-policies
+  namespace: flux-system
+spec:
+  sourceRef:
+    kind: OCIRepository
+    name: kyverno-policies
+  interval: 60m
+  retryInterval: 5m
+  path: ./
+  prune: true
+  wait: true
+  timeout: 2m
+  patches: # enforce all policies
+    - patch: |
+        - op: add
```

Comment:
Why an `add` operation? The field `spec.validationFailureAction` is already present. For a JSONPath this typically results in a `replace` operation.